### DISCUSSION TITLE
Edge attributes in multi-layered GNNs

### DISCUSSION THREAD

**Discussion by antoni4040** [pyg-team/pytorch_geometric#6225] [Q&A]
Hello. I'm doing this for the layers:
```
self.convs = torch.nn.ModuleList([
            TransformerConv(output_features, output_features, 4, False, edge_dim=EDGE_ATTRS_NUMBER)
            for _ in range(num_layers)
        ])
```

And I do this in the forward pass:
```
 for i in self.convs:
            x = i(x.relu(), edge_index, edge_attrs)
            xs.append(x.clone())

```

Now, my problem is that the node data change every time, but the edge_attrs remains the same. Is this the intended way to do? Or maybe should I pass the edge_attrs through a linear layer manually? Thanks

**Comment by rusty1s** [comment]
I think this is the intended way. Each layer will still learn to transform edge features on its own by passing it through a `Linear` layer.
